apps/api/v2/src/modules/slots/slots-2024-09-04/services/slots.service.ts

```diff
@@ -215,6 +223,59 @@ export class SlotsService_2024_09_04 {
     }
   }
 
+  async validateRoundRobinSlotAvailability(
+    eventTypeId: number,
+    startDate: DateTime,
+    endDate: DateTime,
+    hosts: Host[]
+  ) {
+    const fixedHosts = hosts.filter((host) => host.isFixed === true);
+    const nonFixedHosts = hosts.filter((host) => host.isFixed === false);
+
+    if (fixedHosts.length > 0) {
+      await this.validateFixedHostsAvailability(eventTypeId, startDate, endDate, fixedHosts);
+    } else {
+      await this.validateNonFixedHostsAvailability(eventTypeId, startDate, endDate, nonFixedHosts);
+    }
+  }
+
+  async validateFixedHostsAvailability(
+    eventTypeId: number,
+    startDate: DateTime,
+    endDate: DateTime,
+    hosts: Host[]
+  ) {
+    const existingBooking = await this.slotsRepository.getExistingBooking(
+      eventTypeId,
+      startDate.toJSDate(),
+      endDate.toJSDate()
+    );
+    const hasHostAsAttendee = hosts.some((host) =>
```

Comment:
This function is called `validateFixedHostsAvailability` but if there is, for example, a round robin with 1 fixed host and 1 random and we book this event then the fixed , I think the fixed host will not be in the `attendees` but only the non-fixed host will be in `attendees` - the fixed host id will be on `Booking.userId`. If there are 2 fixed hosts then one of them will be in the `attendees` but other will still be on `Booking.userId` so the `const hasHostAsAttendee` would not return true because `Booking.userId` aka fixed host that does not reside in the `attendees` table would be ignored.